docs: add governance.md (#1)
ibis-project/governance: governance.md

project.

These categories are not mutually exclusive.

### Users

Users are people using Ibis.

### Contributors

A contributor is someone who contributes code or documentation to the project.
The primary way someone contributes these artifacts is through GitHub Pull
Requests.

After a period of time, contributors are eligible to become committers.

### Committers

A committer is someone who has the ability to make commits to the project
without the approval of other committers or steering committee members, which
includes the ability to approve and merge pull requests.

**Pull requests should have at least one approving committer, or a stated
reason why a pull request bypassed approval**

To become a committer, a person must show sustained activity for a period of at
least six months and be nominated by a steering committee member.

Committers are responsible for things like:

- Code review
- Ensuring that CI remains green
- Helping on-board new contributors
- Maintaining existing code

Committers who have not made a commit to one or more repositories under the
`ibis-project` organization for a period of one year or longer are subject to

> having their commit privileges removed.

Reviewer: On this I would just say that at least with Apache projects, karma, once granted, can't be taken away. So I think it's fine for Ibis to have the ability to do "committer gardening", but it would be best for this not to be interpreted as threat-like in the sense of "stay active or your commit bit may be taken away". In practice I've rarely seen idle committers doing any harm to a project, or at least doing so would be seen as a faux pas indicative of other problems with such individuals.

Author: Yeah, I'm not going for threatening here. I can reword it to be a bit less "use it or lose it" 😂

The idea with this bit is to be consistent with the notion that folks contributing for recent sustained periods should be given more weight in the current decision making project than folks who have not contributed for an extended period of time, even if they made many contributions during their time on the project.

Reviewer: Yeah, totally agreed with that (the point is not lost on me that I am one of those people in this case 😄). This seems to be the socially-acceptable convention in most projects (e.g. you and I are both pandas committers, but if we showed up right now with Lots Of Opinions About How To Do Things it wouldn't be seen as appropriate), so I don't think there's anything wrong with stating so explicitly.

Author: Clarified, and added a more explicit description

New steering committee members must be nominated by another steering committee member.

The minimum number of votes required to become a steering committee member is
**three**, and only steering committee members' votes are binding.

Nominations should be made and discussed in a [GitHub Discussion](https://github.com/ibis-project/ibis/discussions).

## Decision-making process

### Location

Major decisions should be made **in public** using [GitHub Discussions](https://github.com/ibis-project/ibis/discussions).

Decisions around governance itself should be discussed in
[`ibis-project/governance`'s
discussions](https://github.com/ibis-project/governance/discussions).

### Seeking consensus

Decision-making should be done by seeking consensus from as many community
perspectives as possible.

The community should be given at least one week to respond to a proposed
decision.

Large, sweeping decisions must include the support of at least two steering
committee members.

## AI policy

Any content whose majority is generated by an AI, large language model or
otherwise non-human entity must have a visible disclaimer stating such tools
were used in the creation of the content.

> Use of copilot during day to day development (for example, Python docstring

Reviewer: clarify it's GitHub Copilot?

Author: Good idea, thanks.

Author: Fixed.

> is based on current work and not historic contributions.

Another important point from that model is:

> Major project decisions are made based on a consensus seeking process
> (discuss major grievances) rather than pure vote, and strive to include as
> many community perspectives as possible.

## Project artifacts

The project artifacts are the code and documentation that reside in the
collection of git repositories under the
[`ibis-project`](https://github.com/ibis-project) GitHub organization.

## People

The following categories are loose categories of people involved with the Ibis
project.

These categories are not mutually exclusive.

### Users

Users are people using Ibis.

### Contributors

A contributor is someone who contributes code or documentation to the project.
The primary way someone contributes these artifacts is through GitHub Pull
Requests.

After a period of time, contributors are eligible to become committers.

### Committers

A committer is someone who has the ability to make commits to the project
without the approval of other committers or steering committee members, which
includes the ability to approve and merge pull requests.

> **Pull requests should have at least one approving committer, or a stated

Author: Clarified.

reason why a pull request bypassed approval**

To become a committer, a person must show sustained activity for a period of at
least six months and be nominated by a steering committee member.

Committers are responsible for things like:

- Code review
- Ensuring that CI remains green
- Helping on-board new contributors
- Maintaining existing code

Committers who have not made a commit to one or more repositories under the
`ibis-project` organization for a period of time will be contacted to determine
whether they are still interested in participating in the project and if not,
may have their commit rights removed.

Decision-making power leans towards those people who are currently active in
the project.

#### Becoming a committer

To become a committer, a person must have sustained contributions to one or
more projects in the organization for at least **six months**.

> New committers must be nominated by a steering committee member.
> 
> The minimum number of votes required to become a committer is
> **three**, and only steering committee members' votes are binding.

Reviewer 2: I find these paragraphs a bit unclear (what is the voting process? Is it majority? At least 3 approvals? Full consensus (so any "no" is a blocker)?). I've attempted to clarify it a bit here, feel free to change if my interpretation is incorrect.

```suggestion
Candidates are nominated by a steering committee member, and must
have approval from at least 2 other steering committee members.
```

Author: What you've written is good!

Nominations should be made and discussed in a [GitHub Discussion](https://github.com/ibis-project/ibis/discussions).

### Steering committee members

A steering committee member is a committer that shows commitment to the
long-term health of the Ibis project. In addition to the responsibilities of
committers, committee members are responsible for things like:

- Nominating others for committership and steering committee membership
- Handling security reports in a timely manner
- Enforcing the code of conduct

#### Becoming a steering committee member

Steering committee members must already be committers.

Additionally, steering committee members must have been a committer for at
least six months in addition to the six months required to become a committer
and show commitment to long-term health of the project.

New steering committee members must be nominated by another steering committee member.

The minimum number of votes required to become a steering committee member is

> **three**, and only steering committee members' votes are binding.

Reviewer 2: Same here.

Author: I changed it to be the same as committership.

Nominations should be made and discussed in a [GitHub Discussion](https://github.com/ibis-project/ibis/discussions).

## Decision-making process

### Location

Major decisions should be made **in public** using [GitHub Discussions](https://github.com/ibis-project/ibis/discussions).

Decisions around governance itself should be discussed in
[`ibis-project/governance`'s

> discussions](https://github.com/ibis-project/governance/discussions).

Reviewer 2: Do we want proposals for new committers/project members to be discussed in the `ibis/discussions` (as listed above)? Or `governance/discussions`? I know `ibis/discussions` is listed above, just confirming here.

Author: I think governance/discussions is probably the right place for nomination and discussion. I'll fix.

Author: Fixed.

`ibis-project` organization for a period of time will be contacted to determine
whether they are still interested in participating in the project and if not,
may have their commit rights removed.

Decision-making power leans towards those people who are currently active in
the project.

#### Becoming a committer

To become a committer, a person must have sustained contributions to one or
more projects in the organization. There is no explicit duration over which
this must happen.

Candidates are nominated by a steering committee member, and must
have approval from at least 2 other steering committee members.

Nominations should be made and discussed in a [GitHub Discussion](https://github.com/ibis-project/ibis/discussions).

### Steering committee members

A steering committee member is a committer that shows commitment to the
long-term health of the Ibis project. In addition to the responsibilities of
committers, committee members are responsible for things like:

- Nominating others for committership and steering committee membership
- Handling security reports in a timely manner
- Enforcing the code of conduct

The current steering committee members are:

> - Phillip Cloud (@cpcloud)
> - Krisztián Szűcs (@kszucs)
> - Jim Crist-Harif (@jcrist)
> - Gil Forsyth (@gforsyth)

Reviewer 3: Is the order of the names meaningful? If yes, please explain. If no, please sort alphabetical by last name.
```suggestion
- Phillip Cloud (@cpcloud)
- Jim Crist-Harif (@jcrist)
- Gil Forsyth (@gforsyth)
- Krisztián Szűcs (@kszucs)
```

Author: The order was not meaningful, thanks!

# Ibis project governance

Ibis's governance is inspired by many open source projects that have come
before it, including but not limited to Apache Arrow, Substrait, Dask and
many others.

The closest concrete model to Ibis's governance is the [**liberal contribution** model](https://opensource.guide/leadership-and-governance/#what-are-some-of-the-common-governance-structures-for-open-source-projects).

A major point in that model is:

> the people who do the most work are recognized as most influential, but this
> is based on current work and not historic contributions.

> The next, and perhaps equally important point is:

Reviewer 4: ```suggestion
Another important point from that model is:
```

We should strive to avoid equivocation in governance docs except where the subject at hand is inherently fuzzy.

Author: True, and thank you.

> Major project decisions are made based on a consensus seeking process
> (discuss major grievances) rather than pure vote, and strive to include as
> many community perspectives as possible.

In addition to the people that work on the Ibis project, the project's
artifacts are the code and documentation that reside in the collection of git

> repositories under the `ibis-project` GitHub organization.

Author: I think I addressed this. Let me know if you want me to remove the text entirely.

## People

The following categories are loose categories of people involved with the Ibis
project.

These categories are not mutually exclusive.

### Users

Users are people using Ibis.

### Contributors

A contributor is someone who contributes code or documentation to the project.
The primary way someone contributes these artifacts is through GitHub Pull
Requests.

After a period of time, contributors are eligible to become committers.

### Committers

A committer is someone who has the ability to make commits to the project
without the approval of other committers or steering committee members, which
includes the ability to approve and merge pull requests.

To become a committer, a person must show sustained activity for a period of at
least six months and be nominated by a steering committee member.

Committers are responsible for things like:

- Code review
- Ensuring that CI remains green
- Helping on board contributors
- Maintaining existing code

Committers who have made a commit to one or more repositories under the
`ibis-project` organization for a period of one year or longer are subject to

> having their commit privileges removed.

Author: Heh, pretty import negation :)

# Ibis project governance

Ibis's governance is inspired by many open source projects that have come
before it, including but not limited to Apache Arrow, Substrait, Dask and

> many others.

Reviewer 5: would remove the second "many" here (very minor nit)

Author: Fixed.

collection of git repositories under the
[`ibis-project`](https://github.com/ibis-project) GitHub organization.

## People

The following categories are loose categories of people involved with the Ibis
project.

These categories are not mutually exclusive.

### Users

Users are people using Ibis.

### Contributors

A contributor is someone who contributes code or documentation to the project.
The primary way someone contributes these artifacts is through GitHub Pull
Requests.

After a period of time, contributors are eligible to become committers.

### Committers

A committer is someone who has the ability to make commits to the project
without the approval of other committers or steering committee members, which
includes the ability to approve and merge pull requests.

**Pull requests should have at least one approving committer, or a stated
reason why a pull request bypassed approval**

To become a committer, a person must show sustained activity for a period of at
least six months and be nominated by a steering committee member.

Committers are responsible for things like:

- Code review
- Ensuring that CI remains green

> - Helping on board contributors

Reviewer 5: on board -> onboard?

Author: Fixed!
